src/cocotb/triggers.py

```diff
@@ -759,16 +759,38 @@ def __repr__(self) -> str:
         return fmt.format(type(self).__qualname__, self.name, _pointer_str(self))
 
 
-class _Join(
-    Trigger,
-    Generic[T],
-    metaclass=_ParameterizedSingletonGPITriggerMetaclass,
-):
-    """Fires when a :class:`~cocotb.task.Task` completes."""
+class TaskComplete(Trigger, Generic[T]):
```

Comment:
Could leave for later cleanup, but this could use the ResultType from task module.